### PR TITLE
test(android_alarm_manager_plus): Add integration tests to alarm manager

### PR DESCRIPTION
**Pull request by LukeMoody01**
## Description

This PR adds integration tests to the android_alarm_manager_plus plugin and also adds these tests into the CI job on an android emulator.

## Related Issues

This PR was previously opened under #1113 but had to be closed due to the PR no longer receiving commits from the head branch. Apologies for any inconvenience!

This PR addresses open issue #1103 

@miquelbeltran could you link this to both the issue and the previous PR? I am not able to do that

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by miquelbeltran** [comment]
this integration test is cursed :) I triggered a re-run of the job.

**Comment by LukeMoody01** [comment]
I am not too sure why the callback test is failing in the job. I have tested it a lot locally, even creating new emulators to test on and it seems to work every time.

Any suggestions here would be great 🙂 (As I don't want to bombard the CI workflow by constantly pushing)

FYI @miquelbeltran

File: packages/android_alarm_manager_plus/example/integration_test/android_alarm_manager_plus_test.dart [hunk 1]
```diff
@@ -2,19 +2,227 @@
...
+          // 10 seconds is the smallest amount of time we need to wait for
+          // the callback to happen
+          await Future.delayed(const Duration(seconds: 10));
+          await tester.pumpAndSettle();
+
+          expect(
+            find.byWidgetPredicate(
+              (Widget widget) =>
+                  widget is Text &&
+                  widget.data!.startsWith('Alarm fired 1 times'),
```
Comment (nohli) [inline packages/android_alarm_manager_plus/example/integration_test/android_alarm_manager_plus_test.dart:145]: You could try adding `skipOffstage: false,` for checking if it has something to do with the screen size when using the emulator runner (`reactivecircus/android-emulator-runner@v2`) in our GitHub workflow.
It also works locally for me.

Comment (LukeMoody01) [reply packages/android_alarm_manager_plus/example/integration_test/android_alarm_manager_plus_test.dart:145]: Yup I can give this a try.

Comment (LukeMoody01) [reply packages/android_alarm_manager_plus/example/integration_test/android_alarm_manager_plus_test.dart:145]: Hmm no luck.

**Comment by miquelbeltran** [comment]
Hey! Due to changes in our contributor guidelines, all PRs should now:

- Have a title that follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/) format. e.g. `feat(package_name)!: feature description`
- **DO NOT** modify the CHANGELOG.md or the version in the `pubspec.yaml`. This is going to be an automated process now.

We would ask you kindly to update the PR following these changes.

Thanks!

**Comment by miquelbeltran** [comment]
The test definitely works locally but fails on CI. I think we can merge this, skip the test, and maybe someday go back to it.

**Review by miquelbeltran — APPROVED** [review]

**Comment by miquelbeltran** [comment]
Merged, thanks for the work @LukeMoody01 !

**Pull request merged**